packages/connections/src/modules/telemetry.ts

```diff
@@ -0,0 +1,108 @@
+import { ConnectionInfo, DataService } from 'mongodb-data-service';
+import { createLoggerAndTelemetry } from '@mongodb-js/compass-logging';
+import { isAtlas, isLocalhost, isDigitalOcean } from 'mongodb-build-info';
+import { getCloudInfo } from 'mongodb-cloud-info';
+import ConnectionString from 'mongodb-connection-string-url';
+
+const { track, debug } = createLoggerAndTelemetry('COMPASS-CONNECT-UI');
+
+async function getConnectionData({
+  connectionOptions: { connectionString, sshTunnel },
+}: ConnectionInfo): Promise<Record<string, unknown>> {
+  const connectionStringData = new ConnectionString(connectionString);
+  const hostName = connectionStringData.hosts[0];
+  const { isAws, isAzure, isGcp } = await getCloudInfo(hostName).catch(
+    (err: Error) => {
+      debug('getCloudInfo failed', err);
+      return {};
+    }
+  );
+  const isPublicCloud = isAws || isAzure || isGcp;
+  const publicCloudName = isAws
+    ? 'AWS'
+    : isAzure
+    ? 'Azure'
+    : isGcp
+    ? 'GCP'
+    : '';
+  const authMechanism = connectionStringData.searchParams.get('authMechanism');
+  const authType = authMechanism
+    ? authMechanism
+    : connectionStringData.username
+    ? 'DEFAULT'
+    : 'NONE';
+
+  return {
+    is_localhost: isLocalhost(hostName),
+    is_atlas: isAtlas(hostName),
+    is_public_cloud: !!isPublicCloud,
+    is_do: isDigitalOcean(hostName),
```

Comment:
apparently `isDigitalOcean` and `isAtlas` won't work with an hostname, also we decided to rename them to `is_atlas_url` and `is_do_url` here, and to readd a different and more precise `is_atlas` later

Reply:
after merging #2605 I kept `isAtlas` prop when a connection is established. it will be great if we merge this now and create a follow up. this has been hanging for quite a while now.